### PR TITLE
Throw when component is missing

### PR DESCRIPTION
**Pull request by brigand**
React has a similar warning. Ran into it when working on #3

### CODE REVIEW

**Comment by vadimdemedes** [comment]
Shouldn't this be an error, rather than a warning? Otherwise the previous `Cannot read property 'prototype' of undefined` error is still going to appear after warning.

**Comment by brigand** [comment]
@vadimdemedes fixed.

File: lib/h.js [hunk 1]
```diff
@@ -4,6 +4,10 @@ const flatten = require('lodash.flattendeep');
 const VNode = require('./vnode');
 
 module.exports = (component, props, ...children) => {
+	if (typeof component !== 'function' && typeof component !== 'string') {
+		throw new TypeError(`ink.h: expected component to be a function, but received "${component}". You may have forgotten to export a component.`);
```
Comment (vadimdemedes) [inline lib/h.js]: Could you remove `ink.h` and start with `Expected`?

Also, replace `"${component}"` with `typeof component`.

File: test/h.js [hunk 2]
```diff
@@ -163,3 +163,10 @@ test.failing('don\'t merge children of components', t => {
 
 	t.deepEqual(node.props.children, ['x', 'y']);
 });
+
+test('warns on invalid type', t => {
+	t.throws(() => {
```
Comment (vadimdemedes) [inline test/h.js]: => `t.throws(() => h(), 'Expected component to be a function, got undefined. You may have forgotten to export a component.');`

File: test/h.js [hunk 3]
```diff
@@ -163,3 +163,10 @@ test.failing('don\'t merge children of components', t => {
 
 	t.deepEqual(node.props.children, ['x', 'y']);
 });
+
+test('warns on invalid type', t => {
```
Comment (vadimdemedes) [inline test/h.js]: `warns` => `throws`
`invalid type` => `missing component`

File: lib/h.js (hunk 1 above)
Comment (brigand) [inline lib/h.js]: Fixed

File: test/h.js (hunk 3 above)
Comment (brigand) [inline test/h.js]: Fixed

File: test/h.js (hunk 2 above)
Comment (brigand) [inline test/h.js]: Fixed

**Comment by vadimdemedes** [comment]
Thanks!

**Pull request merged**